### PR TITLE
Added check for  in Eureka status and health url construction (#1795)

### LINKED ISSUE #1795: EurekaClientAutoConfiguration#eurekaInstanceConfigBean incorrectly configures statusPageUrl and healthCheckUrl
If the application is setting `management.context-path` the statusPageUrl and healthCheckUrl are constructed without it if you don't specifically override like so:
```
eureka:
  instance:
    statusPageUrlPath: ${management.context-path}/info
    healthCheckUrlPath: ${management.context-path}/health
```

example:

application.yml
```
management:
  port: 9090
  context-path: /manage
```

The resulting eureka instance config data is showing:
```
...
<statusPageUrl>http://mymachine:9090/info</statusPageUrl>
<healthCheckUrl>http://mymachine:9090/health</healthCheckUrl>
...
```
It would be good if EurekaClientAutoConfiguration#eurekaInstanceConfigBean picked up the managementPort automatically and configured the urls properly:
```
...
<statusPageUrl>http://mymachine:9090/manage/info</statusPageUrl>
<healthCheckUrl>http://mymachine:9090/manage/health</healthCheckUrl>
...
```

### PR DESCRIPTION
**Pull request by strelok1**
@spencergibb Try 2, git still insisted on rewriting the whole file though but at least the history is ok now.

fixes #1795

### CODE REVIEW

File: spring-cloud-netflix-eureka-client/src/main/java/org/springframework/cloud/netflix/eureka/EurekaClientAutoConfiguration.java [hunk 1]
```diff
@@ -16,14 +16,14 @@
 
 package org.springframework.cloud.netflix.eureka;
 
-import java.lang.annotation.Documented;
-import java.lang.annotation.ElementType;
-import java.lang.annotation.Retention;
-import java.lang.annotation.RetentionPolicy;
-import java.lang.annotation.Target;
-
+import com.netflix.appinfo.ApplicationInfoManager;
```
Comment (spencergibb) [inline spring-cloud-netflix-eureka-client/src/main/java/org/springframework/cloud/netflix/eureka/EurekaClientAutoConfiguration.java]: Please don't rearrange imports like this.

File: spring-cloud-netflix-eureka-client/src/main/java/org/springframework/cloud/netflix/eureka/EurekaClientAutoConfiguration.java [hunk 2]
```diff
@@ -77,198 +78,201 @@
...
-		@ConditionalOnMissingClass("org.springframework.cloud.context.scope.refresh.RefreshScope")
-		static class MissingClass {
-		}
-
-		@ConditionalOnMissingBean(RefreshAutoConfiguration.class)
-		static class MissingScope {
-		}
-
-	}
+    private ConfigurableEnvironment env;
```
Comment (spencergibb) [inline spring-cloud-netflix-eureka-client/src/main/java/org/springframework/cloud/netflix/eureka/EurekaClientAutoConfiguration.java]: Again, tabs, not spaces.

**Comment by strelok1** [comment]
@spencergibb ok I think I got my IntelliJ settings in line with the way spring-cloud-netflix is expecting them to be. I also added a simple .editorconfig file to hint to IDEs with support of the formatting style (tabs).

**Comment by strelok1** [comment]
@ryanjbaxter sure, removed

**Comment by spencergibb** [comment]
Merged via 34350cb7ecf581bcc78f89adc1a2565c0030e42d

**Pull request closed without merging**